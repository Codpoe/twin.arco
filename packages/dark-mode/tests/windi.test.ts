import { Processor } from 'windicss/lib/index';
import plugin from '@/windi';

describe('Windi Arco Dark Mode Plugin', () => {
  it('should generate arco dark mode selector', () => {
    const processor = new Processor({ darkMode: false, plugins: [plugin] });
    const result = processor.interpret('bg-red-900 dark:bg-red-100');
    expect(result.styleSheet.build()).toMatchSnapshot();
  });

  it('should generate media dark selector', () => {
    const processor = new Processor({ darkMode: 'media', plugins: [plugin] });
    const result = processor.interpret('bg-red-900 dark:bg-red-100');
    expect(result.styleSheet.build()).toMatchSnapshot();
  });

  it('should generate class dark selector', () => {
    const processor = new Processor({ darkMode: 'class', plugins: [plugin] });
    const result = processor.interpret('bg-red-900 dark:bg-red-100');
    expect(result.styleSheet.build()).toMatchSnapshot();
  });
});
