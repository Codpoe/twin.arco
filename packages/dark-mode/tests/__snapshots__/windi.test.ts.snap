// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Windi Arco Dark Mode Plugin should generate arco dark mode selector 1`] = `
".bg-red-900 {
  --tw-bg-opacity: 1;
  background-color: rgba(127, 29, 29, var(--tw-bg-opacity));
}
body[arco-theme=dark] .dark\\\\:bg-red-100 {
  --tw-bg-opacity: 1;
  background-color: rgba(254, 226, 226, var(--tw-bg-opacity));
}"
`;

exports[`Windi Arco Dark Mode Plugin should generate class dark selector 1`] = `
".bg-red-900 {
  --tw-bg-opacity: 1;
  background-color: rgba(127, 29, 29, var(--tw-bg-opacity));
}
.dark .dark\\\\:bg-red-100 {
  --tw-bg-opacity: 1;
  background-color: rgba(254, 226, 226, var(--tw-bg-opacity));
}"
`;

exports[`Windi Arco Dark Mode Plugin should generate media dark selector 1`] = `
".bg-red-900 {
  --tw-bg-opacity: 1;
  background-color: rgba(127, 29, 29, var(--tw-bg-opacity));
}
@media (prefers-color-scheme: dark) {
  .dark\\\\:bg-red-100 {
    --tw-bg-opacity: 1;
    background-color: rgba(254, 226, 226, var(--tw-bg-opacity));
  }
}"
`;
